dch/public/app/components/watch-form/watch-form.js

```diff
@@ -0,0 +1,205 @@
+                alert("Please select a watch.");
+            } 
+            else if (startTime == null || startTime == '') {
+                alert("Please select a start time.");
+            } 
+            else if (interval == null || interval == '') {
+                alert("Please select a time interval.")
+            }
+            else {
+                console.log("Requesting data.");
+                console.log($scope.selectedWatch);
+            }
+            requestWatchData();
+        }
+
+       /**
+        * @desc - This function populates the start times for a 
+        *         selected watch at 5 minute intervals.
+        *         
+        * @param - watch {object} - the selected watch.
+        */
+
+        function populateTimes(watch) {
+            var milliseconds = watch.TIMESTAMP; 
+            for (var i=0; i<288; i++) {
+                var dateFromMilliseconds = new Date(milliseconds);
+                $scope.times[i] = dateFromMilliseconds.toString();
+                milliseconds = milliseconds + 300000;
+            }
+            console.log($scope.times);
+        }
+
+       /**
+        * @desc - This callback function is called when $http service completes
+        *         its request for total connected devices successfully.
+        *
+        * @param response {object} - Response from the server.
+        */
+    	function successCallback(response) {
+    		$scope.watches = response.data;
+    		console.log($scope.watches);
+    	}
+
+       /**
+        * @desc - This callback function is called when $http service completes
+        *         its request for total connected devices with an error response.
+        *
+        * @param response {object} - Response from the server.
+        */
+    	function errorCallback(response) {
+    		console.log(response);
+    	}
+
+       /**
+        * @desc - This function is used to request the all the data from the 
+        *         DCH server. Once the request completes, a callback is
+        *         executed based on whether or not the request was successful.
+        */
+    	function requestData() {
+    		var responsePromise = $http.get(GET_DATA_QUERY_PATH);
+
+   			responsePromise.then(successCallback, errorCallback);
+    	}
+
+        // Request data from the DCH server.
+    	requestData();
+
+        /**
+         * @desc - This function parses out the query string and requests 
+         *         the data from the DCH server. The watch id, start time,
+         *         and interval is provided by the user. Stop time is the
+         *         current date.       
+         */
+        function requestWatchData() {
+
+            var watch_id = removeSpaces($scope.selectedWatch.id);   
+
+            var start_time = new Date($scope.selectedWatch.startTime);
+            start_time = start_time.getTime();
+
+            var stop_time = new Date();
+            stop_time = stop_time.getTime();
+
```

Comment:
Above variables should be in camel case. Only use the underscore formatting when populating a JSON.

E.g.

```
var data = {
  watch_id: '08734ac034',
  acc_x: 3.42,
  acc_y: 4.23,
  acc_z: 2.11,
  timestamp: 1409823434
}